### PR TITLE
Added mesh visualization and edge element finding

### PR DESCRIPTION
**Pull request by jploveless**
Standalone functions for now. I'll work on integrating them into celeri.py and/or celeri_vis.py

### CODE REVIEW

File: celeri/get_mesh_edge_elements.py [hunk 1]
```diff
@@ -0,0 +1,56 @@
+def get_mesh_edge_elements(meshes):
+    # Find indices of elements lining top, bottom, and sides of each mesh
+    
+	from ismember import ismember
```
Comment (brendanjmeade) [inline celeri/get_mesh_edge_elements.py:4]: @jploveless I'm not able to do this import "from ismember import ismember".  I don't think it's part of the celeri environment.yml.  I'm wondering if there's something built into numpy that might do this.  Perhaps something in the discussion here: https://stackoverflow.com/questions/25923027/matlab-ismember-function-in-python

File: celeri/get_mesh_edge_elements.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	
+	celeri.get_ordered_edge_nodes(meshes)
+	
+	for i in range(meshes):
```
Comment (brendanjmeade) [inline celeri/get_mesh_edge_elements.py:8]: Throughout we're going with an indentation of 4 spaces at each level.  It looks like more here?

File: celeri/plot_mesh.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+def plot_mesh(meshes, fill_value, ax):
+    import matplotlib.collections
+	for i in range(meshes):
```
Comment (brendanjmeade) [inline celeri/plot_mesh.py:3]: Indentation should match line above.

File: celeri/plot_mesh.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+		x_coords = meshes[i].meshio_object.points[:,0]
+		y_coords = meshes[i].meshio_object.points[:,1]
+		vertex_array = np.asarray(meshes[i].verts)
+
+		if not ax: ax=plt.gca()
+		xy = np.c_[x_coords, y_coords]
+		verts= xy[vertex_array]
+		pc = matplotlib.collections.PolyCollection(verts, edgecolor="none", cmap="rainbow")
+		pc.set_array(fill_value)
+		ax.add_collection(pc)
```
Comment (brendanjmeade) [inline celeri/plot_mesh.py:13]: This looks really slick!

File: celeri/get_mesh_edge_elements.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+		coords = meshes[i].meshio_object.points
+		vertices = meshes[i].verts
+		# Arrays of all element side node pairs
+		side_1 = np.sort(np.vstack((vertices[:, 0], vertices[:, 1])).T, 1)
+		side_2 = np.sort(np.vstack((vertices[:, 1], vertices[:, 2])).T, 1)
+		side_3 = np.sort(np.vstack((vertices[:, 2], vertices[:, 0])).T, 1)
+		# Sort edge node array
+		sorted_edge_nodes = np.sort(meshes[i].ordered_edge_nodes, 1)
+		# Indices of element sides that are in edge node array
+		side_1_in_edge, side_1_in_edge_idx = ismember(sorted_edge_nodes, side_1, 'rows')
```
Comment (brendanjmeade) [inline celeri/get_mesh_edge_elements.py:18]: We're using double quotes throughout for all strings.  One way to enforce this is to install the "black" autoformatter (https://formulae.brew.sh/formula/black) locally and run this before submitting.  I followed these instructions (https://dev.to/adamlombard/how-to-use-the-black-python-code-formatter-in-vscode-3lo0) to have VSCode automatically format all .py files using the black autoformatter any time I save them.  This does not apply to ipynb files.

File: celeri/plot_mesh.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+def plot_mesh(meshes, fill_value, ax):
+    import matplotlib.collections
+	for i in range(meshes):
+		x_coords = meshes[i].meshio_object.points[:,0]
+		y_coords = meshes[i].meshio_object.points[:,1]
+		vertex_array = np.asarray(meshes[i].verts)
+
+		if not ax: ax=plt.gca()
```
Comment (brendanjmeade) [inline celeri/plot_mesh.py:8]: Clever!

**Review by brendanjmeade — COMMENTED** [review]
@jploveless Thanks for sharing this.  This is a great PR and this is a big step.  I've added a few comments that I hop might be useful.  Happy to discuss!

File: celeri/get_mesh_edge_elements.py (hunk 1 above)
Comment (jploveless) [reply celeri/get_mesh_edge_elements.py]: I tried some of those (or from another SO post) but the the [ismember package](https://pypi.org/project/ismember/) seemed to be more straightforward in comparing whole rows and just returning row indices, which was important.

File: celeri/get_mesh_edge_elements.py (hunk 2 above)
Comment (jploveless) [reply celeri/get_mesh_edge_elements.py]: Edited (saved with black formatting) in a876f1233bb3b0e1b63337fd2e8b9800816d7e6c

File: celeri/plot_mesh.py (hunk 3 above)
Comment (jploveless) [reply celeri/plot_mesh.py:3]: Edited (saved with black formatting) in a876f1233bb3b0e1b63337fd2e8b9800816d7e6c

File: celeri/get_mesh_edge_elements.py (hunk 5 above)
Comment (jploveless) [reply celeri/get_mesh_edge_elements.py]: Edited (saved with black formatting) in a876f1233bb3b0e1b63337fd2e8b9800816d7e6c

File: celeri/get_mesh_edge_elements.py (hunk 1 above)
Comment (brendanjmeade) [reply celeri/get_mesh_edge_elements.py]: Understood and I just added ismember as a pip import to environment.yml!

**Comment by jploveless** [comment]
Thanks for all these suggestions @brendanjmeade! black auto formatting took care of most of them, and I do think the ismember package is pretty nice to have, but I mainly think that because it replicates the behavior I'm used to from Matlab!

**Comment by brendanjmeade** [comment]
@jploveless Merged!

**Pull request merged**